test/functional/interface_rest.py

```diff
@@ -285,9 +285,9 @@ def run_test(self):
 
         # Make 3 tx and mine them on node 1
         txs = []
-        txs.append(self.nodes[0].sendtoaddress(not_related_address, 11))
-        txs.append(self.nodes[0].sendtoaddress(not_related_address, 11))
-        txs.append(self.nodes[0].sendtoaddress(not_related_address, 11))
+        txs.append(self.nodes[0].sendtoaddress(not_related_address, 11, timeout=120))
+        txs.append(self.nodes[0].sendtoaddress(not_related_address, 11, timeout=120))
+        txs.append(self.nodes[0].sendtoaddress(not_related_address, 11, timeout=120))
```

Comment:
rpc calls don't have a timeout argument.

Also, this test properly failed with an uninitialized read:

```
 node0 stderr ==27834== Thread 7 b-httpworker.2:

==27834== Conditional jump or move depends on uninitialised value(s)

==27834==    at 0x204563: rest_blockhash_by_height(HTTPRequest*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) (rest.cpp:611)

==27834==    by 0x20B4C8: std::_Function_handler<bool (HTTPRequest*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&), bool (*)(HTTPRequest*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)>::_M_invoke(std::_Any_data const&, HTTPRequest*&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) (std_function.h:301)

==27834==    by 0x3B90B4: operator() (std_function.h:706)

==27834==    by 0x3B90B4: HTTPWorkItem::operator()() (httpserver.cpp:55)

==27834==    by 0x3B988D: WorkQueue<HTTPClosure>::Run() (httpserver.cpp:114)

==27834==    by 0x3B5968: HTTPWorkQueueRun(WorkQueue<HTTPClosure>*, int) (httpserver.cpp:342)

==27834==    by 0x3BA1CA: __invoke_impl<void, void (*)(WorkQueue<HTTPClosure> *, int), WorkQueue<HTTPClosure> *, int> (invoke.h:60)

==27834==    by 0x3BA1CA: __invoke<void (*)(WorkQueue<HTTPClosure> *, int), WorkQueue<HTTPClosure> *, int> (invoke.h:95)

==27834==    by 0x3BA1CA: _M_invoke<0, 1, 2> (thread:234)

==27834==    by 0x3BA1CA: operator() (thread:243)

==27834==    by 0x3BA1CA: std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (*)(WorkQueue<HTTPClosure>*, int), WorkQueue<HTTPClosure>*, int> > >::_M_run() (thread:186)

==27834==    by 0x662A66E: ??? (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.25)

==27834==    by 0x54876DA: start_thread (pthread_create.c:463)

==27834==    by 0x6FCD88E: clone (clone.S:95)

==27834== 

{
   Temporary suppression of uninitialized read in rest code
   Memcheck:Cond
   fun:_ZL24rest_blockhash_by_heightP11HTTPRequestRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE
} 
```

Not sure how that could have happened, so you might want to add the suppression for now to the valgrind.supp file

Reply:
Hrmm, that is super weird. Valgrind false positives are very rare, but I'm unable to reproduce. How the heck was this triggered? :)